### DISCUSSION TITLE
Possible to iterate over one query set but show the other?

### DISCUSSION THREAD

**Discussion by creyD** [encode/django-rest-framework#8744] [Question & Answer]
We have a config object in our code and a simple table for config aliases (alias + config). I want to create a generic API using `ModelSerializers` for listing the following for each ALIAS:

    {
        "alias": "test",
        "config_param_1": 1,
        "config_param_2": True
    }

How I imagined the serialiser to look like working with the standard ModelViewSet is like this:

    class DeviceConfigSerializer(serializers.ModelSerializer):
        class Meta:
            model = Alias
            fields = [f.name for f in Config._meta.get_fields() if f.name not in ['id', 'other_fields']]

Then I would need to overwrite the method for getting the fields to take `alias.conf` fields. Trying a couple of ways (like below) I can't quite get it to work.

    def to_representation(self, instance):
        if isinstance(instance, Config):
            return super().to_representation(instance.conf)
        return super().to_representation(instance)

Also tried overwriting `many_init`, with no success.

**TL;DR**: I want the ListSerializer to iterate over the ALIAS model but really show the details for the CONFIG model plus the alias.id.

**Comment by creyD** [comment]
This would easily work if I made the Serializer class the Config instead of the Alias. I always get a ImproperlyConfigured error message. Manually the following overwrite works: 

```python
def to_representation(self, instance):
        return {
            'id': instance.id,
            'config_param_1': instance.conf.config_param_1.name,
            'config_param_2': instance.conf.config_param_2
        }
```

**Comment by creyD** [comment]
Also I got it to work now by explicitly adding the fields like this: `config_param_2 = serializers.CharField(source='conf.config_param_2')`
